--- cfitsio/Makefile.in.orig	2016-01-08 18:27:12.000000000 +0100
+++ cfitsio/Makefile.in	2016-03-17 16:48:57.669945464 +0100
@@ -66,9 +66,8 @@
 		pliocomp.c fits_hcompress.c fits_hdecompress.c \
 		simplerng.c
 
-ZLIB_SOURCES =	zlib/adler32.c zlib/crc32.c zlib/deflate.c zlib/infback.c \
-		zlib/inffast.c zlib/inflate.c zlib/inftrees.c zlib/trees.c \
-		zlib/uncompr.c zlib/zcompress.c zlib/zuncompress.c zlib/zutil.c
+ZLIB_SOURCES =	\
+		zlib/zcompress.c zlib/zuncompress.c
 
 SOURCES = ${CORE_SOURCES} ${ZLIB_SOURCES} @F77_WRAPPERS@
 
@@ -101,7 +100,7 @@
 shared: lib${PACKAGE}${SHLIB_SUFFIX}
 
 lib${PACKAGE}${SHLIB_SUFFIX}: ${OBJECTS}
-		${SHLIB_LD} ${LDFLAGS} -o ${CFITSIO_SHLIB} ${OBJECTS} -lm ${LIBS}
+		${SHLIB_LD} ${LDFLAGS} -o ${CFITSIO_SHLIB} ${OBJECTS} -lm -lz ${LIBS}
 		@if [ "x${CFITSIO_SHLIB_SONAME}" != x ]; then \
 			ln -sf ${CFITSIO_SHLIB} ${CFITSIO_SHLIB_SONAME}; \
 			ln -sf ${CFITSIO_SHLIB_SONAME} $@; \
--- cfitsio/cfitsio.pc.in.orig	2011-05-12 20:35:39.000000000 +0200
+++ cfitsio/cfitsio.pc.in	2011-08-22 08:26:36.306040160 +0200
@@ -7,5 +7,5 @@ Name: cfitsio
 Description: FITS File Subroutine Library
 Version: 3.28
 Libs: -L${libdir} -lcfitsio @LIBS@
-Libs.private: -lm
+Libs.private: -lm -lz
 Cflags: -I${includedir}
